src/pal/src/thread/context.cpp

```diff
@@ -477,7 +498,23 @@ void CONTEXTFromNativeContext(const native_context_t *native, LPCONTEXT lpContex
         ASSIGN_INTEGER_REGS
     }
 #undef ASSIGN_REG
-    
+
+#if HAVE_GREGSET_T
+    if (native->uc_mcontext.fpregs == nullptr)
+    {
+        ULONG preservedUpperBits = lpContext->ContextFlags & CONTEXT_FLAGS_HIGH_BITS_MASK;
+
+        // Reset the CONTEXT_FLOATING_POINT bit, but preserve any of the original high 16
+        // bits that may have changed (depending on what CONTEXT_FLOATING_POINT is defined 
+        // as on this architecture).
+        lpContext->ContextFlags &= ~CONTEXT_FLOATING_POINT;
```

Comment:
How about to replace all of this with one line: lpContext->ContextFlags &= ~(CONTEXT_FLOATING_POINT & CONTEXT_REGISTERS_TYPE_MASK);
where CONTEXT_REGISTERS_TYPE_MASK is 0xFF (or it can have platform specific value if needed).

Anyway, this is just a suggestion, feel free to ignore.